modular_splurt/code/modules/client/preferences.dm

```diff
@@ -1195,123 +1200,544 @@
+
+		// Heals the cook, but damages vampires
 		/datum/reagent/consumable/garlic,
+
+		// Heals body part damage
 		/datum/reagent/consumable/sprinkles,
-		/datum/reagent/consumable/enzyme,
+
+		// Increases temperature
 		/datum/reagent/consumable/hot_ramen,
+
+		// Increases temperature
 		/datum/reagent/consumable/hell_ramen,
+
+		// Adds sugar reagent
+		/datum/reagent/consumable/corn_syrup,
+
+		// Adds sugar, heals brute, burn, suffocation, and toxin
+		/datum/reagent/consumable/honey,
+
+		// Causes temporary blindness and blurred vision
 		/datum/reagent/consumable/tearjuice,
+
+		// Causes unconsciousness, breath loss, brain damage, toxin damage, stamina loss, and blurred vision
 		/datum/reagent/consumable/entpoly,
+
+		// Heals brute and burn damage
 		/datum/reagent/consumable/vitfro,
+
+		// Causes electrocution
 		/datum/reagent/consumable/liquidelectricity,
+
+		// Causes forced speech
 		/datum/reagent/consumable/char,
-		/datum/reagent/consumable/laughsyrup,
-		/datum/reagent/consumable/honey, //zad
+
+		// Secret reagent, makes all food max quality
+		/datum/reagent/consumable/secretsauce,
+
+		// Used for making most food
+		/datum/reagent/consumable/enzyme,
+
+		/*
+		 * The following reagents have effects
+		 * But are too mild to warrant blacklisting
+		 *
+		// Increases temperature
+		/datum/reagent/consumable/hot_coco,
+		*/
 	)
 
-	GLOB.genital_fluids_list = list()
+	// Define base list
+	var/list/consumable_list = subtypesof(/datum/reagent/consumable)
 
-	var/list/paths = subtypesof(/datum/reagent/consumable)
-	LAZYADD(paths, list(
+	// Define additional allowed reagents
+	var/list/whitelist_list = list(
+		// Just water
 		/datum/reagent/water,
+
+		// Causes arousal
+		// Allowed for ERP reasons
 		/datum/reagent/drug/aphrodisiac,
-		/datum/reagent/drug/copium,
-		/datum/reagent/blood
-	))
 
-	for(var/path in paths)
-		var/datum/reagent/instance = find_reagent_object_from_type(path)
+		// Causes positive mood bonus
+		// On overdose: Causes negative mood penalty and disgust
+		/datum/reagent/drug/copium/gfluid,
+
+		// Restores blood volume
+		/datum/reagent/blood,
+	)
+
+	// Add whitelisted entries to main list
+	LAZYADD(consumable_list, whitelist_list)
 
+	// Define final list
+	var/list/reagent_list
+
+	// Define final type-based list
+	var/list/reagent_list_paths
+
+	for(var/reagent in consumable_list)
+		// Define reagent
+		var/datum/reagent/instance = find_reagent_object_from_type(reagent)
+
+		// Check if reagent exists
 		if(!instance)
 			continue
-		if(path in blacklisted)
+
+		// Check if reagent is non-liquid
+		if(instance.reagent_state != LIQUID)
+			// Ignore reagent
 			continue
+
+		// Check if reagent is blacklisted
+		if(reagent in blacklisted)
+			// Ignore reagent
+			continue
+
+		// Check if reagent is manually whitelisted
+		if(reagent in whitelist_list)
+			// Add immediately
+			LAZYADD(reagent_list, instance)
+
+			// Skip further processing
+			continue
```

Comment:
Shouldn't these also be added to `GLOB.genital_fluids_paths`?

Reply:
Yes, they should.